### PR TITLE
Pipeline result config map is not deleted, but got radixjob as owner-…

### PR DESCRIPTION
**Pull request by satr**
…reference

### CODE REVIEW

File: pkg/apis/kube/config_maps.go [hunk 1]
```diff
@@ -95,6 +95,16 @@ func (kubeutil *Kube) ApplyConfigMap(namespace string, currentConfigMap, desired
 	return err
 }
 
+// GetConfigMapListForLabels Get a list of ConfigMaps by Label requirements
+func (kubeutil *Kube) GetConfigMapListForLabels(namespace string, requirements ...labels.Requirement) ([]corev1.ConfigMap, error) {
```
Comment (nilsgstrabo) [inline pkg/apis/kube/config_maps.go:99]: Rename to ListConfigMapsWithSelector(namespace string, labelSelector string) ([]corev1.ConfigMap, error)

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
Added a few suggestions

File: pkg/apis/job/job.go [hunk 2]
```diff
@@ -624,3 +616,39 @@ func (job *Job) updateRadixJobStatus(rj *v1.RadixJob, changeStatusFunc func(curr
...
+			err := job.kubeutil.DeleteConfigMap(configMap.GetNamespace(), configMap.GetName())
+			log.Debugf("Deleted the ConfigMap %s in %s", configMap.GetName(), configMap.GetNamespace())
+			if err != nil {
+				log.Errorf("failed to delete ConfigMap %s while garbage collecting config-maps in %s. Error: %v", configMap.GetName(), namespace, err)
+			}
+		}
+	}
+}
+
+func (job *Job) getRadixJobNameSet(namespace string) (map[string]bool, error) {
```
[the remark this answers is not in the document]
Comment (satr) [reply pkg/apis/job/job.go:644]: Fixed

File: pkg/apis/job/kubejob.go [hunk 3]
```diff
@@ -216,19 +217,41 @@ func (job *Job) getJobConditionFromJobStatus(jobStatus batchv1.JobStatus) (v1.Ra
 }
 
 func (job *Job) getRadixJobResult() (*v1.RadixJobResult, error) {
-	jobResult := &v1.RadixJobResult{}
-	resultConfigMap, err := job.kubeutil.GetConfigMap(job.radixJob.GetNamespace(), job.radixJob.GetName())
+	jobNamespace := job.radixJob.GetNamespace()
+	jobName := job.radixJob.GetName()
+	radixJobNameReq := getRadixJobNameSelectorRequirement(jobName)
+	pipelineResultConfigMapReq := getPipelineResultConfigMapSelectorRequirement()
+	resultConfigMaps, err := job.kubeutil.GetConfigMapListForLabels(jobNamespace, *radixJobNameReq, *pipelineResultConfigMapReq)
```
[the remark this answers is not in the document]
Comment (satr) [reply pkg/apis/job/kubejob.go:224]: Fixed

File: pkg/apis/kube/config_maps.go (hunk 1 above)
Comment (satr) [reply pkg/apis/kube/config_maps.go:99]: Renamed

**Review by nilsgstrabo — APPROVED** [review]

**Pull request merged**